### DISCUSSION TITLE
iframe support in the Typography plugin

### DISCUSSION THREAD

**Discussion by RyanRoberts** [tailwindlabs/tailwindcss#7806] [Ideas]
The Typography plugin supports targeting the `video` tag is there any particular reason it doesn't also work with the `iframe` tag?

I was just looking to do the following for a YouTube video embed:

```
class="prose prose-iframe:aspect-video"
```

TIA

**Comment by cannap** [comment]
this will not work cross domain as far i know  youtube needs to allow your site for that 
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Reply by RyanRoberts** [reply]
I'm not sure I follow what you mean, this is only about styling the aspect ratio of the iframe tag. I allow the YouTube domain in iframe.

**Reply by cannap** [reply]
oh sry my mistake i troguth you want to inject css in to the iframe xD

**Reply by RyanRoberts** [reply]
No worries 😆

**Comment by RyanRoberts** [comment]
FWIW this is what I've added to my Tailwind CSS file to do the job for now…

```css
@layer utilities {
	.prose :where(iframe):not(:where([class~="not-prose"] *)) {
		@apply aspect-video w-full h-auto
	}
}
```

**Reply by happytodev** [reply]
Thanks a lot for this tip Ryan, you save my day ;-)

**Reply by lucianobosco** [reply]
Thanks! Just be aware that if you are using a prose breakpoint, such as `prose-xl`, code should be
```css
@layer utilities {
	.prose-xl :where(iframe):not(:where([class~="not-prose"] *)) {
		@apply aspect-video w-full h-auto
	}
}
```
Not sure if there is a way to add this to all breakpoints at once
